packages/node/src/node.ts

```diff
@@ -1 +1,64 @@
-export default class Node {}
+import * as cf from "@counterfactual/cf.js";
+
+import EventEmitter from "eventemitter3";
+
+import { MultisigAddress, StateChannelInfo } from "./channel";
+
+/**
+ * The Node class is the entry point to the @counterfactual/node package.
+ * It's responsible for:
+ * - encapsulating channels and their apps
+ * - connecting with a provided database to persistently store commitments
+ * - registering a signing service
+ */
+export class Node {
+  private channels: Map<MultisigAddress, StateChannelInfo> = new Map();
+
+  // Maps AppInstanceIDs to the EventEmitters sending/receiving events for
+  // the relevant AppInstance.
+  private eventEmitters: Map<string, EventEmitter> = new Map();
+
+  /**
+   * Returns all of the apps installed across all of the channels in the Node.
+   */
+  getApps(): cf.legacy.app.AppInstanceInfo[] {
+    const apps: cf.legacy.app.AppInstanceInfo[] = [];
+    this.channels.forEach(
+      (channel: StateChannelInfo, multisigAddress: string) => {
+        Object.values(channel.appInstances).forEach(appInstance => {
+          apps.push(appInstance);
+        });
+      }
+    );
+    return apps;
+  }
+
+  /**
+   * Opens a connection specifically for this app with the consumer of this
+   * node.
+   * @returns An EventEmitter to emit events related to this app for consumers
+   * subscribing to app updates.
+   */
+  openApp(appInstanceID: string): EventEmitter {
+    const appInstanceEventEmitter = new EventEmitter();
+    this.setupListeners(appInstanceEventEmitter);
+    this.eventEmitters.set(appInstanceID, appInstanceEventEmitter);
+    return appInstanceEventEmitter;
```

Comment:
Why does the appInstanceID matter at all? Why not simply return an event emitter that's correctly set up, without recording app instance ID?

Reply:
This might change with the updated playground work.

Reply:
Ok